pandas/core/arrays/categorical.py

```diff
@@ -379,6 +382,110 @@ def __init__(
         self._dtype = self._dtype.update_dtype(dtype)
         self._codes = coerce_indexer_dtype(codes, dtype.categories)
 
+    @classmethod
+    def from_dummies(cls, dummies: "DataFrame", ordered=None):
```

Comment:
Can you annotate `ordered`, as well as the return type?

Reply:
Is there a preference for how to annotate the self return type? As I understand it, you can do it with a `TypeVar`, with a string, or by using `from __future__ import annotations` (although only in 3.7+).